sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/HttpPipelineAdapter.java

```diff
@@ -63,26 +61,25 @@ public IHttpResponse send(HttpRequest httpRequest) {
             request.setBody(httpRequest.body());
         }
 
-        return httpPipeline.send(request)
-            .flatMap(response -> response.getBodyAsString()
-                .map(body -> {
-                    logAccountIdentifiersIfConfigured(body);
-                    com.microsoft.aad.msal4j.HttpResponse httpResponse = new com.microsoft.aad.msal4j.HttpResponse()
-                        .body(body)
-                        .statusCode(response.getStatusCode());
-                    httpResponse.addHeaders(response.getHeaders().stream().collect(Collectors.toMap(HttpHeader::getName,
-                        HttpHeader::getValuesList)));
-                    return httpResponse;
-                })
-                // if no body
-                .switchIfEmpty(Mono.defer(() -> {
-                    com.microsoft.aad.msal4j.HttpResponse httpResponse = new com.microsoft.aad.msal4j.HttpResponse()
-                        .statusCode(response.getStatusCode());
-                    httpResponse.addHeaders(response.getHeaders().stream().collect(Collectors.toMap(HttpHeader::getName,
-                        HttpHeader::getValuesList)));
-                    return Mono.just(httpResponse);
-                })))
-            .block();
+        HttpResponse response = httpPipeline.sendSync(request, Context.NONE);
+        String body =  response.getBodyAsBinaryData().toString();
+
+        if (CoreUtils.isNullOrEmpty(body)) {
+            logAccountIdentifiersIfConfigured(body);
+            com.microsoft.aad.msal4j.HttpResponse httpResponse = new com.microsoft.aad.msal4j.HttpResponse()
+                .body(body)
+                .statusCode(response.getStatusCode());
+            httpResponse.addHeaders(response.getHeaders().stream().collect(Collectors.toMap(HttpHeader::getName,
+                HttpHeader::getValuesList)));
+            return httpResponse;
+        }
+
+        // if no body
+        com.microsoft.aad.msal4j.HttpResponse httpResponse = new com.microsoft.aad.msal4j.HttpResponse()
+                .statusCode(response.getStatusCode());
+        httpResponse.addHeaders(response.getHeaders().stream().collect(Collectors.toMap(HttpHeader::getName,
+                HttpHeader::getValuesList)));
```

Comment:
Logic can be merged, the non-body case is the same as the body case except there is no call to `.body(body)`.

```java
com.microsoft.aad.msal4j.HttpResponse httpResponse = new com.microsoft.aad.msal4j.HttpResponse()
    .statusCode(response.getStatusCode());

if (!CoreUtils.isNullOrEmpty(body)) {
    logAccountIdentifiersIfConfigured(body);
    httpResponse = httpResponse.body(body);
}
            httpResponse.addHeaders(response.getHeaders().stream().collect(Collectors.toMap(HttpHeader::getName, HttpHeader::getValuesList)));

return httpResponse;
```